tests/integration/targets/ec2_vpc_vpn/tasks/main.yml

```diff
@@ -35,7 +35,62 @@
       name: testcgw
     register: cgw
 
-  - name: create vpn connection, with customer gateway
+  - name: create transit gateway
+    ec2_transit_gateway:
+      description: "Transit Gateway for vpn attachment"
+    register: tgw
+
+  - name: create vpn connection, with customer gateway, vpn_gateway_id and transit_gateway
+    ec2_vpc_vpn:
+      customer_gateway_id: '{{ cgw.gateway.customer_gateway.customer_gateway_id }}'
+      vpn_gateway_id: '{{ vgw.vgw.id }}'
+      transit_gateway_id: '{{ tgw.transit_gateway.transit_gateway_id }}'
+      state: present
+    register: result
+    ignore_errors: true
+
+  - name: assert creation of vpn failed
+    assert:
+      that:
+        - result is failed
```

Comment:
What it moving this as a unit test https://github.com/ansible-collections/community.aws/blob/main/tests/unit/plugins/modules/test_ec2_vpc_vpn.py? This is only a suggestion.

Reply:
Hi @alinabuzachis I would like to merge this PR, so that I can start the work on Validated Content. I need to understand how placebo recording works before I start with the unit test. I will take it up in a separate PR.

Reply:
Works, thanks!

Reply:
I don't like that this **just** checks for failed.  This could throw any exception and pass the test, we should at least be looking for part of an error message that indicates the issue was the VPN not being in a happy state

Reply:
This task throws an error as transit_gateway_id and vpn_gw_id are mutually exclusive. I have included the assertion to check for the error msg.